apps/storybook/stories/components/TopBar.stories.tsx

```diff
@@ -1,10 +1,36 @@
-import { TopBar } from "@thunderstore/components";
+import {
+  AnonymousLink,
+  DownloadIcon,
+  FilterIcon,
+  InstallIcon,
+  LikeIcon,
+  TopBar,
+  TopBarMenu,
+  TopBarMenuButton,
+} from "@thunderstore/components";
 import { ComponentStory, ComponentMeta } from "@storybook/react";
 import React from "react";
 
 const meta = { component: TopBar } as ComponentMeta<typeof TopBar>;
+const noop = () => null;
```

Comment:
In the "real world", we want to keep the prop required, since I don't see why we'd want to have menu buttons that do nothing. Here, on Storybook, the menu buttons don't need to do anything, except look pretty. So I'd keep this as is.

Reply:
I agree here, but I wonder if it ends up slowing prototyping / WIP coding enough to warrant allowing optionality? In those scenarios you'd likely insert a no-op variable manually for each case you don't have a proper click handler for yet, which might be a bit cumbersome.

A custom linter rule / test enforcing it that would only be validated for prod builds (i.e. in the CI) would seem like the best of both worlds to me, but I don't think that's reasonable to arrange with our current toolkit (it'd be entirely doable with some tools we don't have in use though).

So in short, do we think the tradeoff is worth it? My gut feeling is it'd be nice to allow it being optional, based on a lot of experience with frameworks/libraries forcing constraints that might've not made sense for my use case at the end of the day & that being a frustrating experience to work with. There shouldn't be real harm done by having a non-actionable button after all.

Reply:
I realize I'm nearing a slippery slope fallacy here, but... Does a button need to have an icon, or should we make it an optional prop? If I want to have a button that has just an icon and no text, then the label is optional too. And thus we end up with a button that's invisible and does nothing. And mind you, this isn't some generic button component. It's a component for buttons in a dropdown menu used in the "top bar" of the site.

And I believe that having this discussion has already slowed down the prototyping / WIP coding more than requiring the `onClick` ever will for anyone participating on this project.

So I'd keep this as is.

Reply:
Agreed we can keep it as is for this PR, I'd not make it a blocker. Conversation is worth to have for future reference though. I suppose given the lack of linting rules that would catch it it's worth the potential dev frustration to keep it as required even for prototyping uses. Better to require a bit more effort to avoid accidents.